migo1/nba-teams-stats
# React capstone project - Metrics webapp

### MILESTONES

- Use API that allows to get numeric values based on a parameter
- Use design guidelines including:
           - Colors (select one main color for your website).
           - Typography: font face, size, and weight.
           - Layout: composition and space between elements.
          
### Interactions

**Home page**

- When the page loads, the web app shows the list of categories that could be filtered out by some parameter, for example by the category name.
- Along with the category name, you will display some numeric values per category that come from the API.
- When the user clicks (or taps) on a category item, the application navigates to the details page.

**Details page**

- In the details page, the web app retrieves data from the API to show detailed data in the category.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

**Testing requirements**

- Create unit tests for pure functions in your Redux code.
- Create integration tests for your application using the React Testing Library.
- You may need to mock the access to the API so that your tests don't send actual requests.
- You may need to mock the connection to the Redux Store.

Technical requirements

- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- You should filter the data that you retrieve from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) should have a unique route within the SPA.
- The project is be deployed and accessible online using [Netlify](https://www.netlify.com/) .
- Add a link to the online version of your app to your README file.
